pkg/trie/triedb/node.go

```diff
@@ -26,11 +30,48 @@ type (
 	}
 
 	newValueRef struct {
-		hash *common.Hash
+		hash common.Hash
 		Data []byte
 	}
 )
 
+func NewEncodedValue(value nodeValue, partial []byte, childF onChildStoreFn) (codec.EncodedValue, error) {
+	switch v := value.(type) {
+	case inline:
+		return codec.NewInlineValue(v.Data), nil
+	case valueRef:
+		return codec.NewHashedValue(v.hash.ToBytes()), nil
+	case newValueRef:
+		// Store value in db
+		childRef, err := childF(NewNodeToEncode{partialKey: partial, value: v.Data}, partial, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		// Check and get new new value hash
+		switch cr := childRef.(type) {
+		case HashChildReference:
+			if cr.hash == common.EmptyHash {
+				panic("new external value are always added before encoding a node")
+			}
+
+			if v.hash != common.EmptyHash {
+				if v.hash != cr.hash {
+					panic("hash mismatch")
+				}
+			}
+
+			v.hash = cr.hash
```

Comment:
in the prev check you make sure that `v.hash` and `cr.hash` are the same, so is this assign needed?

Reply:
Solved with this [commit](https://github.com/ChainSafe/gossamer/pull/4039/commits/46bfce7a153db5a9c46cae9ab4021addcd6f2562)